Content/Items/Magnet/Weapons.ThunderBeads.cs

```diff
@@ -0,0 +1,280 @@
+			if (!embedded)
+			{
+				Projectile.ownerHitCheck = false;
+				Projectile.damage /= 2;
+				this.target = target;
+				embedded = true;
+			}
+			else
+			{
+				Helper.PlayPitched("Magic/LightningShortest1", 0.5f, Main.rand.NextFloat(0f, 0.2f), target.Center);
+				fade = 1;
+				for (int i = 2; i < cache.Count - 2; i++)
+				{
+					Dust.NewDustPerfect(cache[i] + Main.rand.NextVector2Circular(6,6), ModContent.DustType<GlowLineFast>(), cache[i].DirectionTo(Owner.Center).RotatedByRandom(0.2f) * Main.rand.NextFloat(1, 2), 0, Color.Cyan, 0.4f);
+				}
+			}
+		}
+
+		public override bool? CanHitNPC(NPC target)
+		{ 
+			if (embedded)
+				return target == this.target && ableToHit;
+
+			return base.CanHitNPC(target);
+		}
+
+		public override void SetPoints(List<Vector2> controlPoints)
+		{
+			if (embedded)
+			{
+				Vector2 playerArmPosition = Main.GetPlayerArmPosition(Projectile);
+				for (int i = 0; i < segments + 1; i++)
+				{
+					float lerper = i / (float)segments;
+					controlPoints.Add(Vector2.Lerp(playerArmPosition, target.Center, lerper));
+				}
+			}
+			else
+			{
+				base.SetPoints(controlPoints);
+			}
+		}
+
+		public override void DrawBehindWhip(ref Color lightColor)
+		{
+			DrawPrimitives();
+			if (embedded)
+			{
+				Main.spriteBatch.Begin(default, BlendState.Additive, default, default, default, default, Main.GameViewMatrix.TransformationMatrix);
+				Texture2D bloomTex = ModContent.Request<Texture2D>(AssetDirectory.Keys + "Glow").Value;
+				for (int i = 0; i < cache.Count - 1; i++)
+				{
+					if (i % 3 != 0)
+						continue;
+
+					Main.spriteBatch.Draw(bloomTex, cache[i] - Main.screenPosition, null, Color.White * 0.7f, 0, bloomTex.Size() / 2, fade * 0.4f, SpriteEffects.None, 0f);
+				}
+
+				Main.spriteBatch.End();
+			}
+			Main.spriteBatch.Begin(default, default, default, default, default, default, Main.GameViewMatrix.TransformationMatrix);
+		}
+
+		private void ManageCache()
+		{
+			cache = new List<Vector2>();
+			SetPoints(cache);
+
+			cache2 = new List<Vector2>();
+			for (int i = 0; i < cache.Count; i++)
+			{
+				Vector2 point = cache[i];
+				Vector2 endPoint = embedded ? target.Center : cache[i];
+				Vector2 nextPoint = i == cache.Count - 1 ? endPoint : cache[i + 1];
+				Vector2 dir = Vector2.Normalize(nextPoint - point).RotatedBy(Main.rand.NextBool() ? -1.57f : 1.57f);
+				if (i > cache.Count - 3 || dir == Vector2.Zero)
+					cache2.Add(point);
+				else
+					cache2.Add(point + dir * Main.rand.NextFloat(8) * fade);
+			}
+		}
+
+		private void ManageTrails()
+		{
+			Vector2 endPoint = embedded ? target.Center : cache[segments];
+			trail ??= new Trail(Main.instance.GraphicsDevice, segments + 1, new TriangularTip(4), factor => 16, factor =>
+			{
+				if (factor.X > 0.99f)
+					return Color.Transparent;
+
+				return new Color(160, 220, 255) * fade * 0.1f * EaseFunction.EaseCubicOut.Ease(1 - factor.X);
+			});
+
+			trail.Positions = cache.ToArray();
+			trail.NextPosition = endPoint;
```

Comment:
Might be worth it to put a linebreak between the areas working on trail and trail2 for clarity

Reply:
Fixed